haystack/components/evaluators/results_evaluator.py

```diff
@@ -0,0 +1,107 @@
+from statistics import quantiles
+from typing import Any, Dict
+
+from pandas import DataFrame
+from pandas import concat as pd_concat
+
+
+class ResultsEvaluator:
+    def __init__(self, pipeline_name: str, results: Dict[str, Any]):
+        self.results = results
+        self.pipeline_name = pipeline_name
+
+    def individual_aggregate_score_report(self) -> Dict[str, float]:
+        """Calculate the average of the scores for each metric in the results."""
+        return {entry["name"]: sum(entry["scores"]) / len(entry["scores"]) for entry in self.results["metrics"]}
+
+    def comparative_aggregate_score_report(self, other: "ResultsEvaluator"):
```

Comment:
Not really sure this function is providing much value as is?

Unless we provide this in a dataframe where we align the metric names. E.g. if pipeline 1 has metrics A, B and pipeline 2 has metrics B, C, we would have a dataframe with index: [pipeline1, pipeline2] and columns [A, B, C] and we arrange it into a dataframe where we show None values for the metrics that are not shared across all pipelines.

If we keep this function, I would rename as `comparative_score_report`.